src/base/utils/net.cpp

```diff
@@ -38,5 +38,23 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isLoopbackAddress(const QHostAddress &clientAddress)
+        {
+            return clientAddress == QHostAddress::LocalHost
+                    || clientAddress == QHostAddress::LocalHostIPv6
+                    || clientAddress == QHostAddress("::ffff:127.0.0.1");
+        }
+
+        bool isLANAddress(const QHostAddress &clientAddress)
```

Comment:
there are non-trivial computing in this func, maybe it make sense to add the following?
```c++
if (clientAddress.isEqual(QHostAddress::AnyIPv4)) {
}
else {
}
```